moveit_ros/planning/planning_scene_monitor/test/current_state_monitor_test.cpp

```diff
@@ -163,6 +174,28 @@ TEST_F(CurrentStateMonitorTest, IncrementalTimeStamps)
   EXPECT_EQ(js_a_old.position[0], csm->getCurrentState()->getVariablePosition("a-b-joint"));
 }
 
+TEST_F(CurrentStateMonitorTest, NonMonotonicTimeStampsDueToPartialJoints)
+{
+  sensor_msgs::JointState js_b_t2;
+  js_b_t2.name = { "b-c-joint" };
+  js_b_t2.position = { 0.25 };
+  js_b_t2.velocity = { 0.25 };
+  js_b_t2.header.stamp = ros::Time{ 13.0 };
+
+  sendJointStateAndWait(js_a);
+  EXPECT_EQ(js_a.header.stamp, csm->getCurrentStateTime());
+
+  sendJointStateAndWait(js_b);
+  EXPECT_EQ(js_b.header.stamp, csm->getCurrentStateTime())
+      << "older partial joint state was ignored in current state retrieval!";
+  EXPECT_EQ(js_a.header.stamp, csm->getCurrentStateTime("group_a"))
+      << "Group is aware of the timestamp of non-group joints!";
+
+  sendJointStateAndWait(js_b_t2);
```

Comment:
Consider simply updating the existing message:
```suggestion
  js_b.header.stamp = ros::Time{ 13.0 };
  sendJointStateAndWait(js_b);
```

Reply:
Well, only changing the timestamp would fail the test, because if the previous joint value is the same as the newcoming one (see [here](https://github.com/ros-planning/moveit/blob/e7f8a82814b860e0553b193ba3dd57fa10121c76/moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp#L318)), the `update` is never set to `true` inside `CurrentStateMonitor::jointStateCallback`, which causes [this](https://github.com/ros-planning/moveit/blob/e7f8a82814b860e0553b193ba3dd57fa10121c76/moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp#L364) update-callback to be skipped. It causes `CurrentStateMonitorTest::sendJointStateAndWait` to fail in its last check of `EXPECT_EQ(std::future_status::ready, status);` because of the [skipped update-callback](https://github.com/ros-planning/moveit/blob/e7f8a82814b860e0553b193ba3dd57fa10121c76/moveit_ros/planning/planning_scene_monitor/test/current_state_monitor_test.cpp#L104-L110).

I can update the block in `jointStateCallback` so that `update` is set to `true` [here](https://github.com/ros-planning/moveit/blob/e7f8a82814b860e0553b193ba3dd57fa10121c76/moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp#L304-L307) -- i.e. when a joint is received with a new timestamp independent from the previous vs. newcoming joint value. But this is a change of behavior, and would obviously cause more frequent update-callback triggers that are currently skipped in such a case, which may be unwanted. 

What do you think @rhaschke? Should we keep the behavior or change it?

Reply:
You are right. We should keep the behavior. But you could simply change the joint value in the test too.
I just wanted to avoid creating a completely new message from scratch.